packages/core/src/connectors/utilities/aliyun.ts

```diff
@@ -0,0 +1,96 @@
+import { createHmac } from 'crypto';
+
+import { has } from '@silverhand/essentials';
+import got from 'got';
+
+import { ConnectorError } from '../types';
+
+// Aliyun has special excape rules.
+// https://help.aliyun.com/document_detail/29442.html
+const escaper = (string_: string) =>
+  encodeURIComponent(string_)
+    .replace(/\*/g, '%2A')
+    .replace(/'/g, '%27')
+    .replace(/!/g, '%21')
+    .replace(/"/g, '%22')
+    .replace(/\(/g, '%28')
+    .replace(/\)/g, '%29')
+    .replace(/\+/, '%2B');
+
+export const getSignature = (
+  parameters: Record<string, string>,
+  secret: string,
+  method: string
+) => {
+  const canoQuery = Object.keys(parameters)
+    .slice()
+    .sort()
+    .map((key) => {
+      const value = parameters[key];
+      if (typeof value !== 'string') {
+        throw new ConnectorError('Invalid value');
+      }
+
+      return `${escaper(key)}=${escaper(value)}`;
+    })
+    .join('&');
+
+  const stringToSign = `${method.toUpperCase()}&${escaper('/')}&${escaper(canoQuery)}`;
+  return createHmac('sha1', `${secret}&`).update(stringToSign).digest('base64');
+};
+
+export interface PublicParameters {
+  AccessKeyId: string;
+  RegionId?: string;
+  Format?: string;
+  Version?: string;
+  SignatureMethod?: string;
+  Timestamp?: string;
+  Signature?: string;
+  SignatureVersion?: string;
+  SignatureNonce?: string;
+}
+
+const commonParameters = {
```

Comment:
similarly, maybe we could have a const config file for these hardcoded congfigs.

Reply:
`aliyun` is a single util file, let's keep the config inside the file until `aliyun` becomes a directory when necessary in the future?

Reply:
o I was thinking of a config file for all connectors.  Any way your call, not a blocker at all.

Reply:
each connector is designed to be "independant", so keep it inside each dir.